src/MongoDB/WriteConcern.c

```diff
@@ -91,7 +90,11 @@ PHP_METHOD(WriteConcern, __construct)
 	switch(ZEND_NUM_ARGS()) {
 		case 3:
 			if (Z_TYPE_P(journal) != IS_NULL) {
+#ifdef ZEND_ENGINE_3
+				mongoc_write_concern_set_journal(intern->write_concern, Z_TYPE_P(journal) == IS_TRUE);
```

Comment:
This is not correct.

zpp accepts a 'z' here, so the type can be whatever. A truthful value, such as 1, will not match == IS_TRUE, but should still be considered as enabling journalling here.

    $wc = new MongoDB\Driver\WriteConcern(1, 0, 1);
    var_dump($wc->getJournal());

This should return true, but after this change -- it returns false